packages/acot-preset-wcag/src/rules/focus-dialog.ts

```diff
@@ -0,0 +1,69 @@
+import { createRule } from '@acot/core';
+import type { ComputedAccessibleNode } from '@acot/types';
+
+type Options = {};
+
+export default createRule<Options>({
+  type: 'contextual',
+  selector: '[aria-haspopup="dialog"]',
+  meta: {
+    description: 'Move focus to inside dialog or set dialog after trigger',
+    tags: ['wcag2.1a', '2.4.3 Focus Order'],
+    recommended: true,
+  },
+
+  test: async (context, node) => {
+    try {
+      await node.click();
+
+      const activeElementHasParentDialogRole = async () => {
+        if (document.activeElement === null) {
+          return false;
+        }
+
+        const ax = (await (window as any).getComputedAccessibleNode(
+          document.activeElement,
+        )) as ComputedAccessibleNode;
+
+        function findParentDialogRoleAXNode(
+          computedAXNode: ComputedAccessibleNode,
+        ): ComputedAccessibleNode | null {
+          if (computedAXNode.role === 'dialog') {
+            return computedAXNode;
+          }
+
+          return computedAXNode.parent != undefined
+            ? findParentDialogRoleAXNode(computedAXNode.parent)
+            : null;
+        }
+
+        return findParentDialogRoleAXNode(ax)?.role === 'dialog';
+      };
```

Comment:
Unless you have a clear reason, I like to use `const` like any other function.

Reply:
OK, I don't have big reason. I fixed.